tests/Doctrine/Tests/DBAL/Functional/NamedParametersTest.php

```diff
@@ -212,6 +214,35 @@ public function testTicket($query,$params,$types,$expected)
             $result[$k] = array_change_key_case($v, CASE_LOWER);
         }
 
-        self::assertEquals($result, $expected);
+        self::assertEquals($expected, $result);
+    }
+
+    public function namedParametersAfterEscapeProvider()
+    {
+        return [
+            [['param_0' => 'bar', 'param_1' => 'foo', 'find' => '%a%']],
+            [['param_0' => 'bar', 'param_1' => 'foo', 'find' => '%o%']],
+        ];
+    }
+
+    /**
+     * @dataProvider namedParametersAfterEscapeProvider
+     * @param array $params
+     */
+    public function testNamedParametersAfterEscape(array $params)
+    {
+        $escapeChar = $this->_conn->getDatabasePlatform()->quoteStringLiteral('\\');
```

Comment:
@morozov the problem is, PostgreSqlPlatform escapes `\`, but the string in `ESCAPE '…'` must not be escaped for PostgreSql

Reply:
@nickvergessen please see if you can use the API introduced in https://github.com/doctrine/dbal/pull/3013.